paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -572,6 +605,20 @@ internal class PaymentSheetViewModel @Inject internal constructor(
         }
     }
 
+    private fun PaymentSelection.New.canSave(): Boolean {
+        val initializationMode = args.initializationMode
+        val requestedToSave =
+            customerRequestedSave == PaymentSelection.CustomerRequestedSave.RequestReuse
+
+        return when (initializationMode) {
+            is PaymentSheet.InitializationMode.PaymentIntent -> requestedToSave
+            is PaymentSheet.InitializationMode.SetupIntent -> true
+            is PaymentSheet.InitializationMode.DeferredIntent -> {
+                initializationMode.intentConfiguration.mode.setupFutureUse != null || requestedToSave
+            }
+        }
```

Comment:
Marking this as especially important to review. Decides if the newly added card should be saved as the default selection.

Reply:
@tillh-stripe can you also take a look (especially here).